### DISCUSSION TITLE
Python Client Update() failing

### DISCUSSION THREAD

**Discussion by officiallymarky** [supabase/supabase#2855] [Questions] [answered]
I am trying to update all rows in the inventory table to set their balance to 0.

`data = supabase.table("inventory").update({"balance" : "0"}).execute()`

When I do this, I get the following error:

`{'data': {'message': 'Error in $: not enough input'}, 'status_code': 400}`

I tried using "match" as showed in the Javascript client update() example and I got:

`AttributeError: 'FilterRequestBuilder' object has no attribute 'match'`

I tried using `.neq("balance", "0")` filter to see if it required a filter, but still same `not enough input` error.

**Comment by steve-chavez** [comment, marked as answer]
Hey @officiallymarky,

> data = supabase.table("inventory").update({"balance" : "0"}).execute()
> {'data': {'message': 'Error in $: not enough input'}, 'status_code': 400}

Try removing the quotes from `0`.

If the error persist, I suggest to open an issue on: https://github.com/supabase/supabase-py
